profiling/src/profile/mod.rs

```diff
@@ -659,47 +645,60 @@ impl Profile {
             local_root_span_id_label_offset,
         };
 
-        let id = match self.samples.get_index_of(&s) {
-            None => {
-                self.samples.insert(s, values);
-                SampleId::new(self.samples.len() - 1)
+        if let Some(ts) = timestamp {
+            match self.timestamped_samples.get_mut(&s) {
+                None => {
+                    let series = vec![(ts, values)];
+                    self.timestamped_samples.insert(s, series);
+                }
+                Some(series) => series.push((ts, values)),
```

Comment:
This seems like a reasonable solution to the repeated timestamp issue -- do nothing!

The pprof spec doesn't say you need to aggregate stuff, but you usually want to do it to save on memory/space. But storing it as a separate sample even though it maybe could've been aggregated if a timestamp is repeated seems like a reasonable trade-off!

(I think it's worth leaving a comment stating this, in case people wonder about it in the future, since it's a quite subtle detail)

Reply:
Done